models/modelsmodule.cpp

```diff
@@ -477,19 +477,31 @@ ModelsModule::init( SLIInterpreter* )
   kernel()
     .model_manager
     .register_secondary_connection_model< GapJunction< TargetIdentifierPtrRport > >(
-      "gap_junction", /*has_delay=*/false, /*requires_symmetric=*/true );
+      "gap_junction",
+      /*has_delay=*/false,
+      /*requires_symmetric=*/true,
+      /*supports_wfr=*/true );
   kernel()
     .model_manager
     .register_secondary_connection_model< RateConnectionInstantaneous< TargetIdentifierPtrRport > >(
-      "rate_connection_instantaneous", /*has_delay=*/false );
+      "rate_connection_instantaneous",
+      /*has_delay=*/false,
+      /*requires_symmetric=*/false,
+      /*supports_wfr=*/true );
```

Comment:
@jakobj Is there any technical reason why this additional parameter `supports_wfr` is necessary? From looking at `nest::ConnectionManager::deliver_secondary_events` I assume that it prevents that SecondaryEvents with `supports_wfr=false` are send in wfr iterations.

Up to now this was decided in the neuron models `update` function, by putting the sending of SecondaryEvents with `supports_wfr=false` after a `if ( not called_from_wfr_update )` condition. Is there any reason why this is not sufficient anymore or is this new parameter more like a safety net?

Reply:
In 5g we have a specific place in the MPI receive buffer at which each connection looks for its event. Let's assume during a gather operation a new event is written to this position by the source rank, and the target rank delivers it. If we now enter the wfr update, the event is still present in the buffer and a connection that does not support wfr would nevertheless deliver that event. This is why we need to check directly in `deliver_secondary_events` whether a particular connector supports wfr and avoid delivering those events.

Reply:
I see, thank you for the explanation.